pkg/rc/replication_controller.go

```diff
@@ -940,14 +940,15 @@ func (rc *replicationController) finishTransfer(rcFields fields.RC) error {
 		if err != nil {
 			return err
 		}
-	}
 
-	txn.AddNode(rc.nodeTransfer.newNode)
+		// We only want to add an audit record for the new node and set its label if we actually wrote it
+		txn.AddNode(rc.nodeTransfer.newNode)
 
-	labelKey := labels.MakePodLabelKey(rc.nodeTransfer.newNode, rcFields.Manifest.ID())
-	err = rc.podApplicator.SetLabelsTxn(txn.Context(), labels.POD, labelKey, rc.computePodLabels(rcFields))
-	if err != nil {
-		return err
+		labelKey := labels.MakePodLabelKey(rc.nodeTransfer.newNode, rcFields.Manifest.ID())
+		err = rc.podApplicator.SetLabelsTxn(txn.Context(), labels.POD, labelKey, rc.computePodLabels(rcFields))
```

Comment:
I'm realizing there's still a race here, where we might have locked the key but then the other RC already wrote these labels. I think that will resolve itself on its own eventually though, the other RC will notice it doesn't have enough nodes to meet desired and schedule it again (which will be a no-op on intent but fix the labels).

Alternatively i could make this be a check-and-set with the index set to 0 so it will fail if the labels are written already, but then i'd have to either set up a new transaction that doesn't write the labels or i'd have to put it in two separate transactions and set up a state machine of some kind stored in the rc status structure